sycl/source/detail/pi.cpp

```diff
@@ -292,16 +292,22 @@ std::vector<std::pair<std::string, backend>> findPlugins() {
   // env only.
   //
 
+#ifndef __INTEL_PREVIEW_BREAKING_CHANGES
   device_filter_list *FilterList = SYCLConfig<SYCL_DEVICE_FILTER>::get();
+#endif
   ods_target_list *OdsTargetList = SYCLConfig<ONEAPI_DEVICE_SELECTOR>::get();
 
+#ifndef __INTEL_PREVIEW_BREAKING_CHANGES
   // Will we be filtering with SYCL_DEVICE_FILTER or ONEAPI_DEVICE_SELECTOR ?
   // We do NOT attempt to support both simultaneously.
   if (OdsTargetList && FilterList) {
     throw sycl::exception(sycl::make_error_code(errc::invalid),
                           "ONEAPI_DEVICE_SELECTOR cannot be used in "
                           "conjunction with SYCL_DEVICE_FILTER");
   } else if (!FilterList && !OdsTargetList) {
+#else
```

Comment:
Does it make sense to refactor this right away, so it looks cleaner and it will be easier to clean it up later?

My suggestion is to have the following:
```c++
  ods_target_list *OdsTargetList = SYCLConfig<ONEAPI_DEVICE_SELECTOR>::get();
#ifndef __INTEL_PREVIEW_BREAKING_CHANGES
  device_filter_list *FilterList = SYCLConfig<SYCL_DEVICE_FILTER>::get();
  if (OdsTargetList && FilterList) {
    throw ...
  }
  if (FilterList) {
    // existing "else if (FilterList)" branch content
   return PluginNames; // new line to have early exit for simplifying rest of the code
  }
#endif
  if (!OdsTargetList) {
    // we will now have early exit above, so we want enter this branch if FilterList is set
    // existing code for the corresponding branch
  } else {
    // suggested code for the corresponding branch (existing + esimd emulator  handling removal under ifdef)
  }
```

This way, there will be only one huge `#ifdef` section, which can easily be removed later.

Reply:
Thanks. I've fixed that.